Sources/PriorityQueue/PriorityQueue.swift

```diff
@@ -0,0 +1,227 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+//Indicates a min or max heap
+public enum HeapType{
+    case max;
+    case min;
+}
+
+/*
+ *  Implements a PriorityQueue using an array-based min/max binary heap.
+ */
+public struct PriorityQueue<Element:Comparable>{
+    
+    //The underlying storage for the min/max heap
+    private var storage:[Element];
+    
+    //Indicates the type of binary heap "storage" is
+    private let storageType:HeapType;
+    
+    //initialize instance variables
+    //eventually add a constructor that accepts a Sequence and generates the underlying storage in O(n) instead of O(nlogn)
+    public init(priorityQueueType:HeapType){
+        storage = [Element]()
+        storageType = priorityQueueType
+    }
+    
+    /*
+     *  Adds an element to the Priority Queue
+     *  @param  value  the element to add to the Priority Queue
+     */
+    public mutating func enqueue(value:Element){
+        storage.append(value)
+        swimUp(elementIndex: storage.count-1);
+    }
+    
+    /*
+     *  Moves a specified element down the array (towards index zero) until it's in its natural position
+     *  @param  elementIndex  the index of the specified element
+     */
+    private mutating func swimUp(elementIndex:Int){
```

Comment:
This package follows [The Leading Underscore Rule](https://github.com/apple/swift/blob/main/docs/StandardLibraryProgrammersManual.md#the-leading-underscore-rule) from the stdlib naming conventions; internal or private members need to have a leading underscore somewhere in their fully qualified name.

```suggestion
    private mutating func _raise(from index: Int) {
```